packages/ra-core/src/dataProvider/useCreate.ts

```diff
@@ -127,7 +127,10 @@ export const useCreate = <
             unknown
         > & { returnPromise?: boolean } = {}
     ) => {
-        const { returnPromise, ...reactCreateOptions } = createOptions;
+        const {
+            returnPromise = options.returnPromise,
```

Comment:
please test and document that option

Reply:
The `returnPromise` option already existed. However we only accepted at call time. I just added it at declaration time